src/fontra/client/core/fontra-menus.js

```diff
@@ -0,0 +1,222 @@
+import * as html from "./html-utils.js";
+import { translate } from "/core/localization.js";
+import { MenuBar } from "/web-components/menu-bar.js";
+
+const mapMenuItemKeyToFunction = {
+  File: getFileMenuItems,
+  Font: getFontMenuItems,
+  // "Edit": getEditMenuItems,
+  // "View": getViewMenuItems,
+  // "Glyph": getGlyphMenuItems,
+};
+
+export function makeFontraMenuBar(menuItemKeys, delegate) {
+  const menuBarArray = [getFontraMenuItems()]; // Fontra-Menu at the beginning.
+
+  for (const itemKey of menuItemKeys) {
+    const methodName = `get${itemKey}MenuItems`;
+    if (typeof delegate[methodName] === "function") {
+      menuBarArray.push(delegate[methodName]());
+    } else if (mapMenuItemKeyToFunction[itemKey]) {
+      menuBarArray.push(mapMenuItemKeyToFunction[itemKey](delegate));
+    } else {
+      console.log("Method/Function does not exist, skip: ", itemKey, methodName);
+    }
+  }
+
+  menuBarArray.push(getHelpMenuItems()); // Help-Menu at the end.
+  const menuBar = new MenuBar(menuBarArray);
+  return menuBar;
+}
+
+function getFontraMenuItems() {
+  return {
+    title: "Fontra",
+    bold: true,
+    getItems: () => {
+      const menuItems = [
+        "shortcuts",
+        "theme-settings",
+        "display-language",
+        "clipboard",
+        "editor-behavior",
+        "plugins-manager",
+        "server-info",
+      ];
+      return menuItems.map((panelID) => ({
+        title: translate(`application-settings.${panelID}.title`),
+        enabled: () => true,
+        callback: () => {
+          window.open(`/applicationsettings/applicationsettings.html#${panelID}-panel`);
+        },
+      }));
+    },
+  };
+}
+
+function getHelpMenuItems() {
+  return {
+    title: translate("menubar.help"),
+    getItems: () => {
+      return [
+        {
+          title: translate("menubar.help.homepage"),
+          enabled: () => true,
+          callback: () => {
+            window.open("https://fontra.xyz/");
+          },
+        },
+        {
+          title: translate("menubar.help.documentation"),
+          enabled: () => true,
+          callback: () => {
+            window.open("https://docs.fontra.xyz");
+          },
+        },
+        {
+          title: translate("menubar.help.changelog"),
+          enabled: () => true,
+          callback: () => {
+            window.open("https://fontra.xyz/changelog.html");
+          },
+        },
+        {
+          title: "GitHub",
+          enabled: () => true,
+          callback: () => {
+            window.open("https://github.com/googlefonts/fontra");
+          },
+        },
+      ];
+    },
+  };
+}
+
+function getFileMenuItems(delegate) {
+  return {
+    title: translate("menubar.file"),
+    getItems: () => {
+      let exportFormats =
+        delegate.fontController?.backendInfo.projectManagerFeatures["export-as"] || [];
```

Comment:
The "delegate" is meant to be an abstraction that *only* provides optional methods to populate menus. This now assumes that it is an object with a "fontController" propery. Either we should let go of the "delegate" abstraction, and just call the thing "viewController", or we need to solve this differently. Let me think.

Reply:
Ok, I thought about it. Let's bite the bullet and rename "deletage" to "viewController" everywhere in this module.